vignettes/miaViz.Rmd

```diff
@@ -326,6 +326,128 @@ names(metadata(dmn_se))
 plotDMNFit(dmn_se, type = "laplace")
 ```
 
+# Serial data ordination and trajectories
+
+```{r, error=FALSE, warning=FALSE, results='hide'}
+if (!requireNamespace("devtools", quietly = TRUE))
+    install.packages("devtools")
+devtools::install_github("microbiome/miaTime")
+```
+
+Principal Coordinate Analysis using Bray-Curtis dissimilarity on the
+`hitchip1006` dataset:
+
+```{r}
+library(dplyr)
+data(hitchip1006, package = "miaTime")
+tse <- hitchip1006
+tse <- transformCounts(tse, method = "relabundance")
+## Ordination with PCoA with Bray-Curtis dissimilarity
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray",
+                  name = "PCoA_BC", exprs_values = "relabundance",
+              na.rm = TRUE)
+# Calculating Explained variance
+e <- attr(reducedDim(tse, "PCoA_BC"), "eig")
+rel_eig <- e/sum(e[e>0])
+# plot
+p1 <- plotReducedDim(tse, dimred = "PCoA_BC") +
+    labs(x = paste("PCoA 1 (", round(100 * rel_eig[[1]],1), "%", ")", sep = ""),
+         y = paste("PCoA 2 (", round(100 * rel_eig[[2]],1), "%", ")", sep = ""),
+         title="PCoA - Bray-Curtis")+
+    theme(plot.title = element_text(hjust = 0.5))
+p1
+```
+
+Retrieving information about all available trajectories:
+
+```{r}
+colData(tse)$time <- as.factor(colData(tse)$time)
+table(table(tse$subject)) %>% as.data.frame() %>%
+    rename(Timepoints=Var1, Subjects=Freq)
+```
+
+Lets look at all trajectories having two time points in the data:
+
+```{r}
+# plot
+# List subjects with two time points
+selected.subjects <- names(which(table(tse$subject)==2))
+
+# plot
+p1 <- plotReducedDim(tse, dimred = "PCoA_BC") +
+         labs(title="PCoA - Bray-Curtis")+
+    theme(plot.title = element_text(hjust = 0.5))
+
+# plot
+df <- reducedDim(tse)
+colnames(df) <- paste0("PC", 1:2)
+df <- cbind(df, colData(tse)) %>% as.data.frame()
+
+p1 <- p1 + geom_path(aes(x=PC1, y=PC2, group=subject),
+              arrow=arrow(length = unit(0.1, "inches")),
+              data = subset(df, subject %in% selected.subjects))+ 
+    labs(title = "All trajectories with two time points")
+
+print(p1)
+```
+
+Filtering by trajectory length and displaying top 10%:
+
+```{r}
+# calculating trajectory lengths as sum of euclidean distances (based on the 
+# ordination coordinates) of each successive time points in each series.
+df <- merge(df,
+               df %>% group_by(subject) %>% group_modify(~{
+                   # distance matrix between points based on PC1 and PC2
+                   mat <- as.matrix(dist(.x[,c("PC1", "PC2")]));
+                   # summing up only the distance between successive points
+                   data.frame(change=sum(mat[row(mat) == col(mat) + 1]))
+               }),
+            by.y="subject") %>% arrange(time)
```

Comment:
I still think that doing it with `stepwiseDivergence` introduces more code and run time for a simple task in this example. Perhaps when `stepwiseDivergence` is updated by then?

Reply:
This is good discussion; I just realized that it would be more truthful to show the distances calculated based on the original data space. In some cases these could differ from distances on the ordination map (which always is a compromise), and being able to spot that when it happens can be very useful.

Hence, I suggest that stepwiseDivergence is used to calculate divergences in the original data and then those divergences are the ones that will be used to color the arrows in the ordination figure.

Would that make sense?

Reply:
of course, if the calculated divergences in the original data are to be used, then I will use the function and color the arrows correspondingly. I will do so next.